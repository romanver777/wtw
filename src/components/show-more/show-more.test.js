/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from "react";
import TestRenderer from "react-test-renderer";
import ShowMore from "./show-more";

describe("ShowMore component", () => {
  const handler = jest.fn();
  const TRenderer = TestRenderer.create(<ShowMore onHandleClick={handler} />);

  it("renders correctly", () => {
    expect(TRenderer.toJSON()).toMatchSnapshot();
  });
  it("should find 'button' with className", () => {
    expect(TRenderer.root.findByType("button").props.className).toBe(
      "catalog__button"
    );
  });
  it("should find click func prop", () => {
    expect(TRenderer.root.findByType("button").props.onClick());
  });
});
