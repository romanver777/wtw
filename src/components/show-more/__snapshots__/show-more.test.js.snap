// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowMore component renders correctly 1`] = `
<div
  className="catalog__more"
>
  <button
    className="catalog__button"
    onClick={[Function]}
    type="button"
  >
    Больше
  </button>
</div>
`;
